src/k8s-extension/azext_k8s_extension/partner_extensions/Dapr.py

```diff
@@ -0,0 +1,67 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+# pylint: disable=unused-argument
+# pylint: disable=line-too-long
+# pylint: disable=too-many-locals
+
+# from typing import Tuple
+from azext_k8s_extension.partner_extensions.DefaultExtension import DefaultExtension
+
+from azure.cli.core.azclierror import InvalidArgumentValueError
+
+from .DefaultExtension import DefaultExtension
+
+from ..vendored_sdks.models import (
+    Extension,
+    ScopeCluster,
+    Scope
+)
+
+
+class Dapr(DefaultExtension):
+    def __init__(self):
+        # constants for configuration settings.
+        self.DEFAULT_RELEASE_NAMESPACE = 'dapr-system'
+        self.CLUSTER_TYPE = 'global.clusterType'
+
+    def Create(self, cmd, client, resource_group_name, cluster_name, name, cluster_type, extension_type,
+               scope, auto_upgrade_minor_version, release_train, version, target_namespace,
+               release_namespace, configuration_settings, configuration_protected_settings,
+               configuration_settings_file, configuration_protected_settings_file):
+
+        """ExtensionType 'Microsoft.Dapr' specific validations & defaults for Create
+           Must create and return a valid 'ExtensionInstance' object.
+        """
+        ext_scope = None
+        if scope == 'namespace':
+            raise InvalidArgumentValueError("Invalid scope '{}'.  This extension can be installed "
+                                            "only at 'cluster' scope.".format(scope))
+
+        if not release_namespace:
+            release_namespace = self.DEFAULT_RELEASE_NAMESPACE
+        scope_cluster = ScopeCluster(release_namespace=release_namespace)
+        ext_scope = Scope(cluster=scope_cluster, namespace=None)
+
+        # If release-train is not input, set it to 'stable'
+        if release_train is None:
+            release_train = 'stable'
+
+        if cluster_type.lower() == '' or cluster_type.lower() == 'managedclusters':
+            configuration_settings[self.CLUSTER_TYPE] = 'managedclusters'
+
+        create_identity = False
+        extension_instance = Extension(
+            extension_type=extension_type,
+            auto_upgrade_minor_version=auto_upgrade_minor_version,
+            release_train=release_train,
+            version=version,
+            scope=ext_scope,
+            configuration_settings=configuration_settings,
+            configuration_protected_settings=configuration_protected_settings,
+            identity=None,
+            location=""
+        )
+        return extension_instance, name, create_identity
```

Comment:
Newline at the end of the file

Reply:
Added the newline. Otherwise, Github would report a red flag.